simple-examples/FileIO.hs

```diff
@@ -0,0 +1,164 @@
+{-# OPTIONS_GHC -fno-warn-name-shadowing #-}
+
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE RebindableSyntax #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE TypeInType           #-}
+{-# LANGUAGE TypeApplications     #-}
+{-# LANGUAGE GADTs                #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE FlexibleInstances    #-}
+{-# LANGUAGE StandaloneDeriving     #-}
+
+
+{-|
+Module      : FileIO
+Description : The Linear File IO example from the Linear Haskell paper.
+
+We implement a function that prints the first line of a file.
+
+We do this with the normal file IO interface in base and the linear file IO
+interface in linear-base. With the latter, the protocol for using files is
+enforced by the linear type system. For instance, forgetting to close the file
+will induce a type error at compile time. That is, typechecking proves that all
+opened files are closed at some later point in execution. With the former
+interface, we have code that type checks but will error or cause errors at
+runtime.
+
+-}
+
+
+module FileIO where
+
+import Prelude
+import qualified System.IO as Sys
+import Control.Monad ()
+import Data.Text
+
+-- Linear Base Imports
+import qualified Control.Monad.Linear.Builder as Control
+import Data.Unrestricted.Linear
+import System.IO.Resource
```

Comment:
This should probably be imported qualified. _e.g._ as `Linear` for the sake of this file.